### PR TITLE
Fix Pipeline

### PR DESCRIPTION
**Pull request by kris-singh**
Solves PR #932

### CODE REVIEW

**Comment by kris-singh** [comment]
Do we have to add sklearn as  dependencies.

File: gensim/test/test_sklearn_integration.py [hunk 1]
```diff
@@ -67,5 +68,15 @@ def testCSRMatrixConversion(self):
             self.assertTrue(isinstance(v, six.string_types))
             self.assertTrue(isinstance(k, int))
 
+    def testPipline(self):
+        model = SklearnWrapperLdaModel(id2word=dictionary, num_topics=2, passes=100, minimum_probability=0, random_state=numpy.random.seed(0))
+        text_lda = Pipeline([('model', model)])
```
Comment (tmylk) [inline gensim/test/test_sklearn_integration.py]: Can a pipeline contain two things? From lda to logistic regression would be good. Also could you please add it to the tutorial.

File: docs/notebooks/sklearn_wrapper.ipynb [hunk 2]
```diff
@@ -292,13 +338,90 @@
...
-   "execution_count": null,
+   "execution_count": 9,
    "metadata": {
     "collapsed": true
    },
    "outputs": [],
-   "source": []
+   "source": [
+    "def scorer(estimator, X,y=None):\n",
+    "    goodcm = CoherenceModel(model=estimator, texts= texts, dictionary=estimator.id2word, coherence='c_v')\n",
```
Comment (tmylk) [inline docs/notebooks/sklearn_wrapper.ipynb]: This gridsearch returns exception in the ipynb. Is it possible to have it fixed?

File: gensim/test/test_sklearn_integration.py (hunk 1 above)
Comment (kris-singh) [inline gensim/test/test_sklearn_integration.py]: Do, you mean to say that we use lda as a feature extractor. And then use it to in the logistic regression. I thought of this and modified the transform function accordingly.

**Comment by kris-singh** [comment]
@tmylk Could you have a look at the Travis . I don't understand why is it failing.

**Comment by tmylk** [comment]
Tests fixed by smart_open update

File: gensim/test/test_sklearn_integration.py [hunk 3]
```diff
@@ -67,5 +84,21 @@ def testCSRMatrixConversion(self):
             self.assertTrue(isinstance(v, six.string_types))
             self.assertTrue(isinstance(k, int))
 
+    def testPipline(self):
```
Comment (tmylk) [inline gensim/test/test_sklearn_integration.py]: typo in name of the function

File: gensim/test/test_sklearn_integration.py [hunk 4]
```diff
@@ -67,5 +84,21 @@ def testCSRMatrixConversion(self):
...
+        model = SklearnWrapperLdaModel(num_topics=2, passes=10, minimum_probability=0, random_state=numpy.random.seed(0))
+        vec = CountVectorizer(min_df=10, stop_words='english')
+        rand = numpy.random.mtrand.RandomState(1) # set seed for getting same result
+        cats = ['rec.sport.baseball', 'sci.crypt']
+        data = fetch_20newsgroups(subset='train',
+                                  categories=cats,
+                                  shuffle=True)
+        text_lda = Pipeline([('features', vec),('model', model)])
```
Comment (tmylk) [inline gensim/test/test_sklearn_integration.py]: please add logistic regression to the pipeline to analyse output of the lda

File: gensim/test/test_sklearn_integration.py [hunk 5]
```diff
@@ -67,5 +84,21 @@ def testCSRMatrixConversion(self):
             self.assertTrue(isinstance(v, six.string_types))
             self.assertTrue(isinstance(k, int))
 
+    def testPipline(self):
+        model = SklearnWrapperLdaModel(num_topics=2, passes=10, minimum_probability=0, random_state=numpy.random.seed(0))
+        vec = CountVectorizer(min_df=10, stop_words='english')
+        rand = numpy.random.mtrand.RandomState(1) # set seed for getting same result
+        cats = ['rec.sport.baseball', 'sci.crypt']
+        data = fetch_20newsgroups(subset='train',
```
Comment (tmylk) [inline gensim/test/test_sklearn_integration.py]: there are smaller datasets in test_data folder. downloading a lot of data makes tests run too long

File: gensim/test/test_sklearn_integration.py (hunk 4 above)
Comment (tmylk) [inline gensim/test/test_sklearn_integration.py]: please add it to the test. 
accuracy is not important here. it is about being in compatible format

File: gensim/test/test_sklearn_integration.py (hunk 5 above)
Comment (kris-singh) [inline gensim/test/test_sklearn_integration.py]: @tmylk i was not able to find a dataset that the labels. If you know can you please tell me which one to use.

Comment (tmylk) [inline gensim/test/test_sklearn_integration.py]: a tiny 100k subset of newsgroups would be ok.

**Comment by kris-singh** [comment]
@tmylk All changes made. Ready for merge. Please let me know if further changes are required.

**Comment by kris-singh** [comment]
@tmylk any other issues that will help with nmf that i could possibly look at.

File: gensim/test/test_sklearn_integration.py [hunk 6]
```diff
@@ -86,19 +92,15 @@ def testCSRMatrixConversion(self):
 
     def testPipline(self):
```
Comment (tmylk) [inline gensim/test/test_sklearn_integration.py]: typo in test name

**Comment by tmylk** [comment]
Thanks! The PR looks good.
For completeess, could you please remove the section inappropriately called "Using together with Scikit learn's Logistic Regression". That section doesn't use gensim at all so shouldn't be in the notebook. It's an omission by the original author.

Please put your new Pipeline section instead of it so users can find it faster.

**Comment by kris-singh** [comment]
Changes made. Also the size of the test file is around 300 kb.

**Comment by tmylk** [comment]
Thanks for the new feature!

**Comment by piskvorky** [comment]
@tmylk this PR has multiple coding style and PEP8 issues. **Do not merge PRs that are not ready for merging.**

File: docs/notebooks/sklearn_wrapper.ipynb [hunk 7]
```diff
@@ -233,72 +239,163 @@
...
    "cell_type": "code",
-   "execution_count": 8,
+   "execution_count": 31,
    "metadata": {
     "collapsed": true
    },
    "outputs": [],
    "source": [
-    "from sklearn import linear_model"
+    "def scorer(estimator, X,y=None):\n",
```
Comment (piskvorky) [inline docs/notebooks/sklearn_wrapper.ipynb]: PEP8: space after comma.

File: docs/notebooks/sklearn_wrapper.ipynb [hunk 8]
```diff
@@ -233,72 +239,163 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 35,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "id2word=Dictionary(map(lambda x : x.split(),data.data))\n",
```
Comment (piskvorky) [inline docs/notebooks/sklearn_wrapper.ipynb]: `map` is discouraged -- use comprehensions and generators.

Also, PEP8 -- space after comma, spaces around ` = `.

File: docs/notebooks/sklearn_wrapper.ipynb [hunk 9]
```diff
@@ -233,72 +239,163 @@
...
+      "Negative features: Fame.:-0.30 considered,:-0.10\n",
+      "0.531040268456\n"
      ]
     }
    ],
    "source": [
-    "clf=linear_model.LogisticRegression(penalty='l1', C=0.1) #l1 penalty used\n",
-    "clf.fit(X,data.target)\n",
-    "print_features(clf,vocab)"
+    "model=SklearnWrapperLdaModel(num_topics=15,id2word=id2word,iterations=50, random_state=37)\n",
```
Comment (piskvorky) [inline docs/notebooks/sklearn_wrapper.ipynb]: PEP8: spaces around assignment operator ` = `. Other space/formatting/PEP8 issues further down this file, but this is the last comment.

File: gensim/sklearn_integration/sklearn_wrapper_gensim_ldamodel.py [hunk 10]
```diff
@@ -109,4 +134,4 @@ def partial_fit(self, X):
         if sparse.issparse(X):
             X = matutils.Sparse2Corpus(X)
 
-        self.update(corpus=X)
+        self.update(corpus=X)
```
Comment (piskvorky) [inline gensim/sklearn_integration/sklearn_wrapper_gensim_ldamodel.py]: PEP8: newline at the end of file.

**Pull request merged**